src/components/Chat/Chat.tsx

```diff
@@ -211,14 +211,16 @@ const ChatWithLLM: React.FC<ChatWithLLMProps> = ({
   };
 
   useEffect(() => {
-    const updateStream = (chunk: ChatCompletionChunk) => {
+    const updateStream = async (chunk: ChatCompletionChunk) => {
       let filesContext = "";
+      const vaultDir = await window.electronStore.getVaultDirectoryForWindow();
```

Comment:
Not in the scope of this PR, but this line probably doesnt need to get executed everytime the token streams to the FE